app/src/main/java/edu/wisc/ece/pinpoint/pages/newpin/NewPinFragment.java

```diff
@@ -0,0 +1,137 @@
+                new NewPinFragmentAdapter(this.getChildFragmentManager(),
+                        newpin_tabLayout.getTabCount(), getLifecycle());
+        newpin_viewPager.setAdapter(fragmentAdapter);
+
+        dropButton = requireView().findViewById(R.id.drop_pin_button);
+        dropButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createNewPin();
+            }
+        });
+
+        //Force the screen to scroll to the bottom after the caption text input gets focus.
+        //Post delay is required as the scroll does not do anything if the fragment hasn't been pushed
+        //up first, because the base fragment is too small (no scroll needed)
+        newpin_inputeditlayout.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if(hasFocus) {
+                    newpin_scrollview.postDelayed(
+                            new Runnable() {
+                                @Override
+                                public void run() {
+                                    newpin_scrollview.scrollTo(0, newpin_scrollview.getHeight()*2);
+                                }
+                }, 100);
+                }else {
+
+                }
+            }
+        });
+
+        newpin_tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
+            @Override
+            public void onTabSelected(TabLayout.Tab tab) {
+                newpin_viewPager.setCurrentItem(tab.getPosition());
+            }
+
+            @Override
+            public void onTabUnselected(TabLayout.Tab tab) {
+                // Mandatory override intentionally blank, will not implement onTabUnselected
+            }
+
+            @Override
+            public void onTabReselected(TabLayout.Tab tab) {
+                // Mandatory override intentionally blank, will not implement onTabReselected
+            }
+        });
+
+        newpin_viewPager.registerOnPageChangeCallback(new ViewPager2.OnPageChangeCallback() {
+            @Override
+            public void onPageSelected(int position) {
+                super.onPageSelected(position);
+                //noinspection ConstantConditions
+                newpin_tabLayout.getTabAt(position).select();
+            }
+        });
+
+
+    }
+
+    private void createNewPin() {
+        EditText contentContainer = requireView().findViewById(R.id.newpin_textcontent_editlayout);
+        String content = String.valueOf(contentContainer.getText());
+        String caption = String.valueOf(newpin_inputeditlayout.getText());
+        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+        Pin p = new Pin(caption, user.getUid(), PinType.TEXT, content);
+        p.save();
+        Toast.makeText(getContext(), "Successfully Dropped Pin!", Toast.LENGTH_SHORT).show();
+    }
```

Comment:
I'd also recommend disabling the drop pin button while this method is executing to prevent people from spam dropping pins. Feel free to reference how that is done in the `save` method of `EditProfileFragment.java`.